### PR TITLE
Add Testing user guide outline

### PR DESCRIPTION
**Pull request by ntrogh**
Fixes https://github.com/microsoft/vscode-docs/issues/7208

### CODE REVIEW

**Comment by ntrogh** [comment]
@isidorn I created an outline for the Testing user guide. I'll continue to further complete the doc. Welcome your feedback.

**Comment by isidorn** [comment]
Sorry for the slow response. I just checked this out and I like this outline!
The only part that I think is super important and I do not know if you covered is the "testing getting started". Do we plan to have a paragraph in the intro on how to actually setup testing? 

And I suggest to share with Connor when possible to also get his feedback.

**Comment by isidorn** [comment]
Sounds good. Thank you very much 🙏

File: docs/editor/testing.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+After you run your tests, you can view the test results in the Test Explorer view, editor gutter, and Test Results panel. To diagnose issues with your tests, you can also [run and debug tests](#run-and-debug-tests) and set breakpoints in your test and application code.
+
+You can also [run tests with coverage](#test-coverage) to see how much of your code is covered by your tests. Test coverage helps you identify areas of your code that are not being tested and ensures that your tests are comprehensive. You can view the test coverage results in different views in the editor and Test Coverage panel.
+
+To optimize your testing workflow, you can [create tasks to run your tests](#task-integration), and optionally run your tests in the background with every code change.
+
+## Testing extensions
+
+You can find extensions that support testing by looking in the [Visual Studio Marketplace](https://marketplace.visualstudio.com/search?target=VSCode&category=Testing&sortBy=Installs). You can also go to the Extensions view (`kb(workbench.view.extensions)`), and filter by the **Testing** category. You can then sort by **Install count** or ratings to see which extensions are popular.
```
Comment (isidorn) [inline docs/editor/testing.md:32]: I would remove the "you can then sort by..." sentence. By default they should be sorted in a good way.

**Review by isidorn — DISMISSED** [review]

**Comment by ntrogh** [comment]
@connor4312 I've started working on a user guide about Testing. Appreciate your perspectives on the current article - it's about 80% ready. Is the outline and structure ok, are there things missing or incorrect? Thanks!

File: docs/editor/testing.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+
+Software testing is a crucial aspect of the software development process. It helps ensure that your code works as expected and catches bugs early in the development cycle. Visual Studio Code provides a rich set of features for testing your code. You can automatically discover tests in your project, run and debug your tests, and get test coverage results.
+
+In this article, you'll learn how to get started with testing in VS Code, discover popular testing extensions, and explore the testing capabilities.
+
+![Testing in Visual Studio Code](images/testing/testing-hero.png)
+
+## Getting started with testing in VS Code
+
+Testing support in VS Code is language-specific and depends on the [extensions](#testing-extensions) you have installed. The testing features are provided by the language extensions, which can be language servers, test runners, or other tools that support testing in that language. To get started with testing in VS Code, first install the appropriate language extension for your project.
```
Comment (connor4312) [inline docs/editor/testing.md:20]: Testing can be in language extensions, but not always. E.g. there's no de-facto standard test framework for JavaScript so support is found in standalone extensions.

File: docs/editor/testing.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+<!-- <div class="marketplace-extensions-testing"></div> -->
+
+> **Tip**: The extensions shown above are dynamically queried. Select an extension tile to read the description and reviews to decide which extension is best for you.
+
+## Automatic test discovery in Test Explorer
+
+The Test Explorer view provides a centralized place to manage and run your tests. You can access the Test Explorer view by selecting the beaker icon in the Activity Bar or by using the **Testing: Focus on Test Explorer View** command in the Command Palette (`kb(workbench.action.showCommands)`).
+
+If you have a project with tests, the Test Explorer view automatically discovers and lists the tests in your workspace. By default, the discovered tests are displayed in a tree view in the Test Explorer. The tree view matches the hierarchical structure of your tests, making it easy to navigate and run your tests.
```
Comment (connor4312) [inline docs/editor/testing.md:42]: There's also a little bit of "it depends" here, for example the Python extension requires users to run a configure command before it shows tests. (A button to do so is shown in the welcome view if there are no tests in the workspace yet.) The user might need to consult the readme of their extension to set up tests.

File: docs/editor/testing.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+- In the Test Coverage view
+
+    A tree view shows the tests with their coverage percentage. A color indicator also gives a visual cue about the coverage percentage. Hover over each line to see more details about the coverage results.
+
+    ![Test Coverage view](images/testing/test-coverage-view.png)
+
+- As an overlay in the editor
+
+    For code files that have test coverage, the editor shows a color overlay in the gutter to indicate which lines are covered by tests or not. When you hover over the gutter, notice that for covered lines, there is also an indicator for the number of times the line was executed.
```
Comment (connor4312) [inline docs/editor/testing.md:108]: Should also mention command and the editor action (in the top right of the editor) to "toggle inline coverage" to show the code highlights persistently

File: docs/editor/testing.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+    ![View coverage in Explorer view](images/testing/view-coverage-in-explorer.png)
+
+- In the diff editor
+
+    If you have a diff editor open, the coverage results are also shown in the diff editor, similar to how they're shown in the editor.
+
+## Task integration
+
+Tasks in VS Code can be configured to run scripts and start processes within VS Code, without having to enter a command line or write new code. In VS Code, you can define a default test task that runs your tests, and optionally create keyboard shortcuts to run the tests.
```
Comment (connor4312) [inline docs/editor/testing.md:124]: Currently tasks don't have special integration into VS Code's testing functionality, so running tests in a task won't update VS Code's UI.

File: docs/editor/testing.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+Use the **Show Output** button in the Test Explorer to view the test output in the **Test Results** panel.
+
+![Test Results panel](images/testing/test-results-panel.png)
+
+You can access the commands for the Test Explorer, such as running or debugging tests, via the Command Palette (`kb(workbench.action.showCommands)`). Enter `Test Explorer:` to find the corresponding commands.
+
+## Test coverage
+
+Test coverage is a measure of how much of your code is covered by your tests. It helps you identify areas of your code that are not being tested. VS Code provides built-in support for running tests with coverage and viewing the coverage results.
```
Comment (connor4312) [inline docs/editor/testing.md:92]: We should mention that this also requires the test extension to support coverage.

File: docs/editor/testing.md (hunk 3 above)
Comment (ntrogh) [reply docs/editor/testing.md:42]: Added a note at the end of this section.

File: docs/editor/testing.md (hunk 5 above)
Comment (ntrogh) [reply docs/editor/testing.md:124]: Included this in the note that extension can implement integrated test automation features.

**Review by connor4312 — APPROVED** [review]

**Review by isidorn — APPROVED** [review]

**Pull request merged**